omnigibson/tasks/grasp_task.py

```diff
@@ -0,0 +1,242 @@
+    ):
+        self.obj_name = obj_name
+        self._primitive_controller = None
+        self._reset_poses = None
+        self._objects_config = objects_config
+        if precached_reset_pose_path is not None:
+            with open(precached_reset_pose_path) as f:
+                self._reset_poses = json.load(f)
+        super().__init__(termination_config=termination_config, reward_config=reward_config)
+
+    def _load(self, env):
+        for obj_config in self._objects_config:
+            obj = env.scene.object_registry("name", obj_config["name"])
+            # Create object
+            if obj is None:
+                obj = create_class_from_registry_and_config(
+                    cls_name=obj_config["type"],
+                    cls_registry=REGISTERED_OBJECTS,
+                    cfg=obj_config,
+                    cls_type_descriptor="object",
+                )
+                # Import the object into the simulator and set the pose
+                env.scene.add_object(obj)
+
+            obj_pos = [0.0, 0.0, 0.0] if "position" not in obj_config else obj_config["position"]
+            obj_orn = [0.0, 0.0, 0.0, 1.0] if "orientation" not in obj_config else obj_config["orientation"]
+            obj_pos, obj_orn = T.pose_transform(*env.scene.prim.get_position_orientation(), obj_pos, obj_orn)
+            obj.set_position_orientation(obj_pos, obj_orn)
+
+    def _create_termination_conditions(self):
+        terminations = dict()
+        # terminations["graspgoal"] = GraspGoal(
+        #     self.obj_name
+        # )
+        terminations["timeout"] = Timeout(max_steps=self._termination_config["max_steps"])
+        # terminations["falling"] = Falling()
+
+        return terminations
+
+    def _create_reward_functions(self):
+        rewards = dict()
+        rewards["grasp"] = GraspReward(self.obj_name, **self._reward_config)
+        return rewards
+
+    def _reset_agent(self, env):
+        robot = env.robots[0]
+        robot.release_grasp_immediately()
+
+        # If available, reset the robot with cached reset poses.
+        # This is significantly faster than randomizing using the primitives.
+        if self._reset_poses is not None:
+            joint_control_idx = np.concatenate([robot.trunk_control_idx, robot.arm_control_idx[robot.default_arm]])
+            robot_pose = random.choice(self._reset_poses)
+            robot.set_joint_positions(robot_pose["joint_pos"], joint_control_idx)
+            robot_pos = np.array(robot_pose["base_pos"])
+            robot_orn = np.array(robot_pose["base_ori"])
+            # Move it to the appropriate scene. TODO: The scene should provide a function for this.
+            robot_pos, robot_orn = T.pose_transform(*robot.scene.prim.get_position_orientation(), robot_pos, robot_orn)
+            robot.set_position_orientation(robot_pos, robot_orn)
+
+        # Otherwise, reset using the primitive controller.
+        else:
```

Comment:
This seems oddly specific, and not exactly unique to this task. Does this also apply to other tasks, e.g.: PointNav? @cgokmen you might know best here

Reply:
It could be reused for other tasks but the pre-generation is a bother so I'm not sure. Let's keep it limited to this for now.